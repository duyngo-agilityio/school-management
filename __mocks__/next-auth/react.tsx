import { createContext } from "react";

export const useSession = () => ({
  data: {
    user: {
      token: "access-token",
      email: "user@gmail.com",
      id: "1",
      name: "User",
    },
    expires: Date.now().toString(),
  },
  status: "authenticated",
  update: async () => null,
});

export type Session = {
  data: {
    user: {
      token: string;
      email: string;
      id: string;
      name: string;
    };
    expires: string;
  };
  status: "loading" | "authenticated" | "unauthenticated";
  update: (data?: Session) => Promise<Session | null>;
};

const INITIAL_SESSION: Session = {
  data: {
    user: {
      token: "access-token",
      email: "user@gmail.com",
      id: "1",
      name: "User",
    },
    expires: Date.now().toString(),
  },
  status: "authenticated",
  update: async () => null,
};

const SessionContext = createContext(INITIAL_SESSION);

export const SessionProvider = ({ children }: React.PropsWithChildren) => {
  <SessionContext.Provider value={INITIAL_SESSION}>
    {children}
  </SessionContext.Provider>;
};
