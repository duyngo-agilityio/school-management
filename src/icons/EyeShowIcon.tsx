import React from 'react';

interface EyeShowIconProps {
  width?: string;
  height?: string;
  stroke?: string;
}

const EyeShowIcon = ({
  width = '16',
  height = '16',
  stroke = '#13296A',
}: EyeShowIconProps) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width={width}
      height={height}
      viewBox="0 0 24 24"
      fill="none"
    >
      <path
        d="M1 12C1 12 5 4 12 4C19 4 23 12 23 12"
        stroke={stroke}
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
      <path
        d="M1 12C1 12 5 20 12 20C19 20 23 12 23 12"
        stroke={stroke}
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
      <circle
        cx="12"
        cy="12"
        r="3"
        stroke={stroke}
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  );
};

export default EyeShowIcon;
