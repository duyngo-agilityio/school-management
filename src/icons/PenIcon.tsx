interface PenIconProps {
  width?: string;
  height?: string;
  stroke?: string;
}

const PenIcon = ({
  width = '16',
  height = '16',
  stroke = 'white',
}: PenIconProps) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width={width}
      height={height}
      viewBox="0 0 25 24"
      fill="none"
    >
      <path
        d="M5.5499 15.5999L9.7499 19.1999M12.7499 22.1999H22.3499M4.9499 15.5999L16.7813 3.35533C18.0552 2.08143 20.1206 2.08143 21.3945 3.35533C22.6684 4.62923 22.6684 6.69463 21.3945 7.96853L9.1499 19.7999L3.1499 21.5999L4.9499 15.5999Z"
        stroke={stroke}
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  );
};

export default PenIcon;
