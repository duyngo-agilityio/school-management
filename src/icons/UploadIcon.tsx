interface UploadIconProps {
  width?: string;
  height?: string;
  stroke?: string;
}

const UploadIcon = ({
  width = '36px',
  height = '36px',
  stroke = '#141B34',
}: UploadIconProps) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width={width}
      height={height}
      viewBox="0 0 24 24"
      fill="none"
    >
      <path
        d="M12 4.5L12 14.5M12 4.5C11.2998 4.5 9.99153 6.4943 9.5 7M12 4.5C12.7002 4.5 14.0085 6.4943 14.5 7"
        stroke={stroke}
        stroke-width="1.5"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
      <path
        d="M20 16.5C20 18.982 19.482 19.5 17 19.5H7C4.518 19.5 4 18.982 4 16.5"
        stroke={stroke}
        stroke-width="1.5"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  );
};

export default UploadIcon;
