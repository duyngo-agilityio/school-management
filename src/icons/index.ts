export { default as SearchIcon } from './SearchIcon';
export { default as ProfileAddIcon } from './ProfileAddIcon';
export { default as BankIcon } from './BankIcon';
export { default as TeacherIcon } from './TeacherIcon';
export { default as EyeHidden } from './EyeHidden';
export { default as EyeShow } from './EyeShow';
export { default as BellNotification } from './BellNotification';
