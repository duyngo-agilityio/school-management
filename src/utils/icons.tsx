import { CallIcon, SmsIcon, StudentIcon } from '@/icons';

// The icons used on the information details page
export const createIcons = (
  width = '24',
  height = '24',
  stroke = '#A7A7A7',
) => [
  {
    id: 1,
    icon: <StudentIcon width={width} height={height} stroke={stroke} />,
  },
  {
    id: 2,
    icon: <CallIcon width={width} height={height} stroke={stroke} />,
  },
  {
    id: 3,
    icon: <SmsIcon width={width} height={height} stroke={stroke} />,
  },
];
