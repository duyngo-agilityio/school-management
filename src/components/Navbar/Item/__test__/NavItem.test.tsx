import { render } from '@testing-library/react';

// Components
import NavItem from '..';

// Icons
import { HomeIcon } from '@/icons';

const NavItemProps = {
  icon: <HomeIcon />,
  href: '',
  isActive: false,
  children: <>Dashboard</>,
};

const NavItemComponent = (overrideProps = {}) => {
  const props = { ...NavItemProps, ...overrideProps };

  return render(<NavItem {...props} />);
};

describe('NavItem Component', () => {
  it('should match snapshot for NavItem', () => {
    const { container } = NavItemComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render with isActive prop as true', () => {
    const { container } = NavItemComponent({ isActive: true });
    expect(container).toMatchSnapshot();
  });
});
