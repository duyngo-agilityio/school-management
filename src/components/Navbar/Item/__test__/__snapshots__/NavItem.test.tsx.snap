// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem Component should match snapshot for NavItem 1`] = `
<div>
  <a
    href=""
    style="text-decoration: none;"
  >
    <div
      class="css-up6y35"
      role="group"
    >
      <div
        class="css-70qvj9"
      >
        <div
          class="css-iikgio"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 16 17"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.6666 7.49978C14.6666 6.69312 14.1266 5.65978 13.4666 5.19978L9.34659 2.31312C8.41325 1.65978 6.91325 1.69312 6.01325 2.39312L2.41992 5.19312C1.81992 5.65978 1.33325 6.65312 1.33325 7.40645V12.3464C1.33325 13.8931 2.59325 15.1598 4.13992 15.1598H11.8599C13.4066 15.1598 14.6666 13.8931 14.6666 12.3531V10.2864"
              stroke="white"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M8 12.4932V10.4932"
              stroke="white"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </div>
        Dashboard
      </div>
    </div>
  </a>
</div>
`;

exports[`NavItem Component should render with isActive prop as true 1`] = `
<div>
  <a
    href=""
    style="text-decoration: none;"
  >
    <div
      class="css-1h5qz1l"
      role="group"
    >
      <div
        class="css-70qvj9"
      >
        <div
          class="css-iikgio"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 16 17"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.6666 7.49978C14.6666 6.69312 14.1266 5.65978 13.4666 5.19978L9.34659 2.31312C8.41325 1.65978 6.91325 1.69312 6.01325 2.39312L2.41992 5.19312C1.81992 5.65978 1.33325 6.65312 1.33325 7.40645V12.3464C1.33325 13.8931 2.59325 15.1598 4.13992 15.1598H11.8599C13.4066 15.1598 14.6666 13.8931 14.6666 12.3531V10.2864"
              stroke="white"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M8 12.4932V10.4932"
              stroke="white"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </div>
        Dashboard
      </div>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
      </svg>
    </div>
  </a>
</div>
`;
